include/uv.h

```diff
@@ -431,10 +427,56 @@ UV_EXTERN int uv_send_buffer_size(uv_handle_t* handle, int* value);
 UV_EXTERN int uv_recv_buffer_size(uv_handle_t* handle, int* value);
 
 UV_EXTERN int uv_fileno(const uv_handle_t* handle, uv_os_fd_t* fd);
+UV_EXTERN int uv_dup(uv_os_fd_t fd, uv_os_fd_t* dup);
 
 UV_EXTERN uv_buf_t uv_buf_init(char* base, unsigned int len);
 
 
+#ifdef _MSC_VER
+# define INLINE __inline
+#else
+# define INLINE inline
+#endif
+
+INLINE static uv_os_fd_t uv_get_osfhandle(int fd)
+{
+#ifdef _WIN32
+  /* _get_osfhandle() raises an assert in debug builds if the FD is invalid. */
+  /* So if you plan on using invalid fd, you will need to install a _CrtSetReportHook handler */
+  return (HANDLE) _get_osfhandle(fd);
+#else
+  return fd;
+#endif
+}
+
+
+INLINE static uv_os_fd_t uv_convert_fd_to_handle(int fd)
+{
+#ifdef _WIN32
+  HANDLE new_handle;
+  if (uv_dup((HANDLE) _get_osfhandle(fd), &new_handle))
+    return INVALID_HANDLE_VALUE;
```

Comment:
I was assuming that callers would be expecting a HANDLE (pointer), not an error code (integer). That way they can pass the return value straight into the libuv API and get UV_EBADF, rather than needing to first check for a magic value.